libfreerdp/core/listener.c

```diff
@@ -60,10 +65,71 @@ static BOOL freerdp_listener_open(freerdp_listener* instance, const char* bind_a
 #ifdef _WIN32
 	u_long arg;
 #endif
+	BOOL useVsock = TRUE;
 
 	if (!bind_address)
 		ai_flags = AI_PASSIVE;
 
+	for (int i = 0; i < strlen(bind_address); i++)
+	{
+		if (!isdigit(bind_address[i]) && bind_address[i] != '-')
+		{
+			useVsock = FALSE;
+		}
+	}
+
+	if (useVsock)
+	{
```

Comment:
May I suggest to move this block in its own function...

Reply:
fixed.